skll/config/utils.py

```diff
@@ -170,16 +173,39 @@ def _parse_and_validate_metrics(metrics, option_name, logger=None):
         raise TypeError("{} should be a list, not a {}.".format(option_name,
                                                                 type(metrics)))
 
-    # `mean_squared_error` is no more supported.
-    # It is replaced by `neg_mean_squared_error`
+    # `mean_squared_error` is no longer supported.
+    # It has been replaced by `neg_mean_squared_error`
     if 'mean_squared_error' in metrics:
         raise ValueError("The metric \"mean_squared_error\" "
                          "is no longer supported."
                          " please use the metric "
                          "\"neg_mean_squared_error\" instead.")
 
-    invalid_metrics = [metric for metric in metrics if metric not in SCORERS]
-    if invalid_metrics:
+    # find any metrics that are not built into SKLL;
+    # these are candidates for custom metrics so try to register them
+    possible_custom_metrics = [metric for metric in metrics if metric not in SCORERS]
+    if (len(possible_custom_metrics) > 0 and
+            (custom_file is not None and custom_file != '')):
+
+        # try to register each potential metric
+        for custom_metric in possible_custom_metrics:
+            try:
+                register_custom_metric(custom_file, custom_metric)
+
+            # do not raise an error if the registration fails since
+            # we will raise an exception for all invalid metrics later
+            except AttributeError:
```

Comment:
Good catch! 

1. The `AttributeError` is raised as a result of the `getattr()` call in `register_custom_metric()`, i.e., when you are trying to import a function that does not exist in the file. You are right that it should be listed in the doctoring even though it's note explicitly raised in the code.

2. I will include the other exceptions here. I am a fan of explicitly listing out the exceptions rather than just using the base exception which is generally not recommended.

3. I also like your idea of updating `invalid_metrics`  instead of using `pass`.